common_utils.py

```diff
@@ -11,6 +16,13 @@
 random.seed(42) # for reproducibility
 
 # OUR LLM + EMBEDDING SETUP
-set_seed(42)  # for reproducibility
-generator = pipeline("text-generation", model="gpt2")
-embedder = SentenceTransformer("sentence-transformers/all-MiniLM-L6-v2")
+
+def get_generator(name: str, *args, **kwargs) -> AbstractGenerator:
+    if name == "gpt2":
+        return GPT2Generator(*args, **kwargs)
+    return None
```

Comment:
I guess it should throw an exception if there's no matching generator, otherwise we can end up with some weird bugs.